projects/plugins/boost/app/modules/cache/cache.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+class Boost_Cache {
+	private $cache_key     = false;
+	private $path_key      = false;
+	protected $request_uri = false;
+
+	public function __construct() {
+		$this->request_uri = isset( $_SERVER['REQUEST_URI'] )
+			? $_SERVER['REQUEST_URI'] // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+			: false;
+
+		add_action( 'wp', array( $this, 'sanitize' ), 0 );
+	}
+
+	/*
+	 * Sanitize the request_uri.
+	 *
+	 * This is called on the wp action, because sanitization functions are not available before that.
+	 */
+	public function sanitize() {
+		if ( ! $this->request_uri ) {
+			return;
+		}
+
+		$this->request_uri = esc_url_raw( wp_unslash( $this->request_uri ) );
+	}
+
+	/*
+	 * Returns true if the request is cacheable.
+	 *
+	 * If a request is in the backend, or is a POST request, or is not an
+	 * html request, it is not cacheable.
+	 * The filter boost_cache_cacheable can be used to override this.
+	 *
+	 * @return bool
+	 */
+	public function is_cacheable() {
+		if ( ! apply_filters( 'boost_cache_cacheable', $this->request_uri ) ) {
+			return false;
+		}
+
+		if ( $this->is_backend() ) {
+			return false;
+		}
+
+		if ( ( isset( $_SERVER['REQUEST_METHOD'] ) && $_SERVER['REQUEST_METHOD'] === 'POST' ) ) {
+			return false;
+		}
+
+		if ( isset( $_SERVER['HTTP_ACCEPT'] ) && strpos( $_SERVER['HTTP_ACCEPT'], 'text/html' ) === false ) { // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			return false;
+		}
+
+		return true;
+	}
+
+	/*
+	 * Returns a key to identify the visitor's cache file.
+	 * It is based on the REQUEST_URI, the cookies and the page GET parameters.
+	 *
+	 * @param string $request_uri (optional) The request uri to use to calculate the cache key. Defaults to the current request uri.
+	 *
+	 * @return string
+	 */
+	private function calculate_cache_key( $request_uri = '' ) {
```

Comment:
It seems a bit strange to accept a `$request_uri` parameter here, given this function uses superglobals like `$_COOKIE` and `$_GET`, which are associated with the current request.

If you ever supply an alternate `$request_uri`, the cache key it generates will still use the cookies and parameters from the current request. Is there a situation where we might want that behaviour?

Reply:
You're right. I should pass those as optional parameters too. The idea being, that given a URL, we can pass it to this function to get the cache key MD5. That, along with the correct path, can be used to do admin tasks like deleting associated cache files when a post is edited.

Reply:
I overhauled the parameters, putting them in an $args array, and brought them all the way up to the get() and set() functions.